common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -246,9 +246,24 @@ class ModuleI18nService(object):
     i18n service.
 
     """
+    def __init__(self):
+        self._xBlock_i18n_info = {}
+
     def __getattr__(self, name):
         return getattr(django.utils.translation, name)
 
+    @property
+    def xBlock_i18n_info(self):
+        """
+        :return _xBlock_i18n_info dictionary
+        xBlock_i18n_info = {
+            'root': xBlock root path,
+            'domain': xBlock domain name,
+            'locale': xBlock locale info
+        }
+        """
+        return self._xBlock_i18n_info
```

Comment:
We're not pursuing it, we're iterating :)  @asadiqbal08 and I have discussed and we're all on the same page about having the runtime derive the root in the default case.  It's a bit tedious to get there, however, so in the meantime I want to gather up these other ideas to see how it all looks together.